packages/@aws-cdk/aws-ec2/lib/security-group.ts

```diff
@@ -602,5 +609,5 @@ function egressRulesEqual(a: CfnSecurityGroup.EgressProperty, b: CfnSecurityGrou
  * Whether this rule refers to all traffic
  */
 function isAllTrafficRule(rule: any) {
-  return rule.cidrIp === '0.0.0.0/0' && rule.ipProtocol === '-1';
+  return (['0.0.0.0/0','::/0'].includes(rule.cidrIp) && rule.ipProtocol === '-1';
 }
```

Comment:
Should this check for **all** traffic? Meaning **both** IPv4 and IPv6 allowed? Otherwise this isn't coherent with the (new and fixed) behavior of `allowAllOutbound`... This is used to guard against some problems, I'm not sure if the check must be "as strict" or not. It could be problematic as it appears to limit what users are able to express using the API:
- `allowAllOutbound: true` will allow **both** IPv4 and IPv6
- `allowAllOutbound: false` would **disallow** allowing **only one of** IPv6 xor IPv6

Reply:
You're right, the 'includes' check is too loose.  I will amend it when I have time. I've been very busy with work, hence not getting around to writing any tests yet..

Reply:
An Egress rule Resource includes `CidrIp` for setting IPv4 and  `CidrIpv6` for setting IPv4: 

```json
Type: AWS::EC2::SecurityGroupEgress
Properties: 
  CidrIp: String
  CidrIpv6: String
  Description: String
  DestinationPrefixListId: String
  DestinationSecurityGroupId: String
  FromPort: Integer
  GroupId: String
  IpProtocol: String
  ToPort: Integer

``` 
A rule can only include  CidrIp or CidrIpv6, so there is no way to test both on the same rule, which makes this check kinda of tricky now

Reply:
Did some digging into this, changing the default to allow all IPv6 traffic and all IPv4 traffic requires  changing `isAllTrafficRule` to return true if the rule being added either allow all IPv4 traffic **OR** allow all IPv6 traffic. But doing so means that we don't allow creating a security group that allows  is that we don't allow creating a security group that allows only one of them, as it will throw:

```
'Cannot add an "all traffic" egress rule in this way; set allowAllOutbound=true on the SecurityGroup instead.'
```
When setting `allowAllOutbound` to false and trying to add an all traffic rule for, WLOG IPv4.

The idea behind not allowing all traffic rule to be added using the `addEgressRule` was that these rules will be removed by ec2 if other rules exists, but of course this is only true for same protocol rules.
Additionally, changing the `isAllTrafficRule` to return true if the rule is either all IPv4 or IPv6 will break customers that use the `addEgressRule` method to allow all IPv6 traffic:

```ts
const mySecurityGroup = new ec2.SecurityGroup(this, 'SecurityGroup', {
  vpc,
  description: 'Allow ssh access to ec2 instances',
  allowAllOutbound: true   // Allow all IPv4 traffic
});
mySecurityGroup.addIngressRule(ec2.Peer.anyIpv6(), Port.allTraffic()); // Allow IPv6
```

Reply:
![Screen Shot 2020-07-16 at 8 38 33 PM](https://user-images.githubusercontent.com/8578043/87746053-746bb080-c7a4-11ea-8cad-d4b5abbb4be7.png)